### DISCUSSION TITLE
Way to remove overlays?

### DISCUSSION THREAD

**Discussion by sswatson** [nushell/nushell#14444] [Q&A]
I'm wondering if there's a way to actually remove an overlay, as opposed to just hiding it.

The use case I have in mind is automatically invoking `overlay use pyv` and `overlay hide pyv` to activate/de-activate a Python virtual environment when entering/leaving a directory containing a `.venv` folder. (I have a module called `pyv` that modifies `$env.PATH` and the prompt.)

The problem with doing this in the naive way is that the _second_ time you enter a directory with a Python virtual environment, the invocation of `overlay use pyv` causes a remembered value of `$env.PWD` to be applied, leading to a surprising change of directory.

The workaround I'm using currently using is to `cd $after` after doing `overlay use pyv`. This works but feels pretty hacky.

Note: a closely related issue was posted here: https://github.com/nushell/nushell/issues/9541

**Comment by NotTheDr01ds** [comment]
The fact that the second `overylay use` is changing directories sounds a lot like #14048.

But no, there's no way to truly delete anything from the stack.  I've seen an explanation (or hypothesis, IIRC) as to why it needs to work this way, but I can't recall where that was now.

**Reply by sswatson** [reply]
Thanks for the validation and for the explanation.
